R/reliabilityAnalysisFrequentist.R

```diff
@@ -0,0 +1,603 @@
+                 "Greatest Lower Bound", "Average interitem correlation", "mean", "sd"),
+      tables_item = c("McDonald's \u03C9", "Cronbach's \u03B1", "Guttman's \u03BB2", "Guttman's \u03BB6", 
+                      "Greatest Lower Bound", "Item-rest correlation", "mean", "sd"),
+      coefficients = c("McDonald's \u03C9", "Cronbach's \u03B1", "Guttman's \u03BB2", "Guttman's \u03BB6", 
+                       "Greatest Lower Bound", "Item-rest correlation"),
+      plots = list(expression("McDonald's"~omega), expression("Cronbach\'s"~alpha), expression("Guttman's"~lambda[2]), 
+                   expression("Guttman's"~lambda[6]), "Greatest Lower Bound")
+    )
+  )
+
+  # order to show in JASP
+  derivedOptions[["order"]] <- c(5, 1, 2, 3, 4, 6, 7, 8)
+  
+  return(derivedOptions)
+}
+
+.frequentistReliabilityReadData <- function(dataset, options) {
+  
+  variables <- unlist(options[["variables"]])
+  if (is.null(dataset)) {
+    dataset <- .readDataSetToEnd(columns.as.numeric = variables, columns.as.factor = NULL, exclude.na.listwise = NULL)
+  }
+  return(dataset)
+}
+
+.frequentistReliabilityCheckErrors <- function(dataset, options) {
+  
+  .hasErrors(dataset = dataset, perform = "run",
+             type = c("infinity", "variance", "observations"),
+             observations.amount = " < 3",
+             custom = .checkEigen,
+             exitAnalysisIfErrors = TRUE)
+  
+}
+# check for negative eigenvalues, positive semidefiniteness
+.checkEigen <- function() {
+  if (any(eigen(cov(x))$values < 0)) {
+    return(gettext("Data covariance matrix is not positive semidefinite"))
+  }
+}
+
+
+# estimate reliability ----
+# maybe in the future it would be easier to have one function for every estimator...
+.frequentistReliabilityMainResults <- function(jaspResults, dataset, options) {
+  
+  model <- jaspResults[["modelObj"]]$object
+  relyFit <- model[["relyFit"]]
+  if (is.null(model)) {
+    
+    model <- list()
+    variables <- options[["variables"]]
+    
+    samples <- options[["noSamplesf"]]
+    p <- ncol(dataset)
+    
+    if (length(variables) > 2L) {
+      
+      dataset <- as.matrix(dataset) # fails for string factors!
+      if (length(options[["reverseScaledItems"]]) > 0L) {
+        cols <- match(unlist(options[["reverseScaledItems"]]), .unv(colnames(dataset)))
+        total <- min(dataset, na.rm = T) + max(dataset, na.rm = T)
+        dataset[ ,cols] = total - dataset[ ,cols]
+      }
+
+      # observations for alpha interval need to be speccified: 
+      model[["obs"]] <- nrow(dataset)
+      
+      model[["footnote"]] <- .frequentistReliabilityCheckLoadings(dataset, variables)
+      
+      if (any(is.na(dataset))) {
+        if (options[["missingValuesf"]] == "excludeCasesPairwise") {
+          missing <- "pairwise"
+          use.cases <- "pairwise.complete.obs"
+          model[["footnote"]] <- paste0(gettext(model[["footnote"]], "Using pairwise complete cases. "))
```

Comment:
```suggestion
          model[["footnote"]] <- gettextf("%s Using pairwise complete cases. ", model[["footnote"]])
```

Reply:
Would it be possible to use gettext around the paste0 function? Cause the paste0 function does put out a string, doesn't it, and that could be wrapped in gettext?